cradle-admin-tool-http/src/main/java/com/exactpro/th2/cradle/adm/http/PageManager.java

```diff
@@ -117,8 +148,48 @@ public void run() {
             try {
                 autoPageInfo.setBookInfo(checkBook(autoPageInfo.getBookInfo(), autoPageInfo.getAutoPageConfiguration()));
             } catch (Exception e) {
-                logger.error("Exception processing book {}", bookName, e);
+                LOGGER.error("Exception processing book {}", bookName, e);
             }
         });
     }
+
+    private static BookInfo getOrCreateBook(@NotNull CradleStorage storage, String bookName, boolean autoBook) {
+        try {
+            BookInfo storageBookInfo = storage.refreshBook(bookName);
```

Comment:
If I am not mistaken this method will throw an error if the book does not exist. I don't think that this is the behavior we want.
The only safe method in CradleStorage API I see is `listBooks()` that returns the collection of all books in the DB

Reply:
I checked this method and it doesn't throw exception, but I like your advice

Reply:
Could you please re-check that? Because I am pretty sure that if the book does not exist it will throw an exception.
We call `storage.refreshBook` method. It calls `getBookCache().loadBook` method. There is only one implementation that does anything in this method - `ReadThroughBookCache`:
```java
if (bookEntity == null) {
    throw new CradleStorageException(String.format("Book %s was not found in DB", bookId.getName()));
}
```
I might be wrong but still, it is better to check it one more time

Reply:
yes, you are right storage.refreshBook throws an exception when book doesn't exist